lib/auth/sessions.go

```diff
@@ -37,6 +38,11 @@ import (
 // The certificate is used for all access requests, which is where access
 // control is enforced.
 func (s *Server) CreateAppSession(ctx context.Context, req services.CreateAppSessionRequest, user services.User, checker services.AccessChecker) (services.WebSession, error) {
+	if !modules.GetModules().Features().App {
+		return nil, trace.AccessDenied(
+			"this Teleport cluster doesn't support application access, please contact the cluster administrator")
+	}
+
```

Comment:
This is fine for now, but something I talked about with @alex-kovoy yesterday was adding an endpoint to auth where other processes (like database service or application service) can query and ask what supported features are (or am I running in cloud, etc). That way this check could then just be enforced upon process startup.

Reply:
@russjones Yep, I had the same thought yesterday, would be nice to be able to enforce this during service startup.